### PR TITLE
First blog post for ESTL - git push force-with-lease

### PR DESCRIPTION
**Pull request by weiqingtoh**
This PR: 
- Removes the default jekyll post
- Adds a logo to the about page for our blog
- Adds the first blog post (copied from my own blog)

### CODE REVIEW

**Comment by weiqingtoh** [comment]
@fonglh

**Comment by fonglh** [comment]
So we're going with jekyll instead of hugo? I'm ok, just confirming that it's what we're going to commit to.

File: _posts/2016-05-10-git-force-push.markdown [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+1. Pick up a feature, create local feature branch.
+1. Do the necessary changes, commit to local feature branch.
+1. Push a local branch to the remote, then open a pull request for team members' review.
+1. Update local branch with members' comments and update the pull request.
+1. Merge pull request into `master` branch.
+
+## Force push-ing
+
+In most typical scenarios, this works well. However, code review often result in me having to *amend or squash my commit(s)*. Because of that, we can't do a simple `git push` to update the feature branch; rather we have to do `git push --force` (or `git push -f`).
```
Comment (fonglh) [inline _posts/2016-05-10-git-force-push.markdown]: `often results in me`

File: _posts/2016-05-10-git-force-push.markdown [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+When we do a `git push --force`, this is the first thing and comes to mind:
```
Comment (fonglh) [inline _posts/2016-05-10-git-force-push.markdown]: `that comes to mind`

File: _posts/2016-05-10-git-force-push.markdown [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+![alt text][force-push-logo]
+
+And of course, a horror story: [Jenkins developer accidentally forced pushed to 150+ github repos][jenkins-force-push]. Many have sworn off it, but others do recommend to do it under controlled conditions.
+
+Here is a guide on how to do a proper [force push][force-push-guide].
+
+## A safe alternative: force-with-lease
```
Comment (fonglh) [inline _posts/2016-05-10-git-force-push.markdown]: Should this be 'safer alternative' too? So it's consistent with line 34. Or if it's actually safe, then change line 34.

File: _posts/2016-05-10-git-force-push.markdown [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+A force push overwrites a remote branch with your local branch, **regardless** of the status of that branch (more on force push [here](force-push-docs)). This is not ideal in a team scenario as it might result in a member overwriting other members' commits (especially if the member forgot to do a `git rebase`).
```
Comment (fonglh) [inline _posts/2016-05-10-git-force-push.markdown]: I think it should be forgetting to `git pull` which causes the problem? Cos that's the one which will fetch commits pushed by other team members.

File: _posts/2016-05-10-git-force-push.markdown [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+__Enter the safer alternative__: `git push --force-with-lease`
+
+Force with lease gives you the flexibility to override new commits on your remote branch, whilst protecting your old commit history:
+
+- If you rebased and squashed/edited/created new commits, you will be able to update the remote  branch.
```
Comment (fonglh) [inline _posts/2016-05-10-git-force-push.markdown]: Are there 2 spaces between "remote branch"?

**Comment by weiqingtoh** [comment]
Thanks for the comments! 👍

**Pull request merged**